### PR TITLE
REST Controller

### PR DESCRIPTION
**Pull request by OlegMR0**

### CODE REVIEW

**Comment by OlegMR0** [comment]
Help with create method please.

File: src/main/java/com/example/bookstore/service/BookServiceImpl.java [hunk 1]
```diff
@@ -1,23 +1,43 @@
...
+    public BookDto getById(Long id) {
+        return bookMapper.toDto(bookRepository.getById(id));
     }
 
     @Override
-    public List<Book> findAll() {
-        return bookRepository.findAll();
+    public BookDto findById(Long id) {
+        return bookMapper.toDto(bookRepository.findById(id)
+                .orElseThrow(EntityNotFoundException::new));
```
Comment (okuzan) [inline src/main/java/com/example/bookstore/service/BookServiceImpl.java]: add id to the message

File: src/main/java/com/example/bookstore/service/BookService.java [hunk 2]
```diff
@@ -1,10 +1,16 @@
...
 
 public interface BookService {
-    Book save(Book book);
+    BookDto save(CreateBookRequestDto book);
+
+    List<BookDto> findAll();
+
+    BookDto getById(Long id);
+
+    BookDto findById(Long id);
```
Comment (okuzan) [inline src/main/java/com/example/bookstore/service/BookService.java]: you need only one

File: src/main/java/com/example/bookstore/service/BookServiceImpl.java [hunk 3]
```diff
@@ -1,23 +1,43 @@
...
+                .map(bookMapper::toDto)
+                .toList();
+    }
+
     @Override
-    public Book save(Book book) {
-        return bookRepository.save(book);
+    public BookDto getById(Long id) {
+        return bookMapper.toDto(bookRepository.getById(id));
     }
```
Comment (okuzan) [inline src/main/java/com/example/bookstore/service/BookServiceImpl.java]: use findById instead

File: src/main/java/com/example/bookstore/dto/CreateBookRequestDto.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package com.example.bookstore.dto;
+
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Data
+public class CreateBookRequestDto {
+    private String title;
+    private String author;
+    private BigDecimal price;
```
Comment (okuzan) [inline src/main/java/com/example/bookstore/dto/CreateBookRequestDto.java:10]: add isbn
rebuild project so mapper can accommodate this change
image

File: src/main/java/com/example/bookstore/controller/BookController.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    public BookDto getBookById(@PathVariable Long id) {
+        return bookService.findById(id);
+    }
+
+    //TODO POST doesn't work
+    @PostMapping()
+    public BookDto createBook(@RequestBody CreateBookRequestDto bookDto) {
+        return bookService.save(bookDto);
+    }
+
```
Comment (okuzan) [inline src/main/java/com/example/bookstore/controller/BookController.java]: remove line

File: src/main/java/com/example/bookstore/controller/BookController.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@AllArgsConstructor
+@RequestMapping(value = "/books")
+public class BookController {
+    private final BookService bookService;
+
+    @GetMapping()
```
Comment (okuzan) [inline src/main/java/com/example/bookstore/controller/BookController.java]:
```suggestion
    @GetMapping

```

File: src/main/java/com/example/bookstore/controller/BookController.java [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+        return bookService.findAll();
+    }
+
+    @GetMapping("/{id}")
+    public BookDto getBookById(@PathVariable Long id) {
+        return bookService.findById(id);
+    }
+
+    //TODO POST doesn't work
+    @PostMapping()
```
Comment (okuzan) [inline src/main/java/com/example/bookstore/controller/BookController.java]:
```suggestion
    @PostMapping

```

**Review by okuzan — CHANGES_REQUESTED** [review]
One field was missing

File: pom.xml [hunk 8]
```diff
@@ -33,6 +34,16 @@
...
 
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-web</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-thymeleaf</artifactId>
+        </dependency>
```
Comment (olekskov) [inline pom.xml]: why we need this dependency?

**Pull request merged**